mesonbuild/build.py

```diff
@@ -62,6 +63,73 @@ class DFeatures(TypedDict):
         import_dirs: T.List[IncludeDirs]
         versions: T.List[T.Union[str, int]]
 
+    class BuildTargetKeywordArguments(TypedDict, total=False):
+
+        # The use of Sequence[str] here is intentional to get the generally
+        # undesirable behavior of Sequence[str]
+
+        build_by_default: bool
+        build_rpath: str
+        c_pch: T.Sequence[str]
+        cpp_pch: T.Sequence[str]
+        d_debug: T.List[T.Union[str, int]]
+        d_import_dirs: T.List[IncludeDirs]
+        d_module_versions: T.List[T.Union[str, int]]
+        d_unittest: bool
+        dependencies: T.List[dependencies.Dependency]
+        extra_files: T.List[File]
+        gnu_symbol_visibility: Literal['default', 'internal', 'hidden', 'protected', 'inlineshidden', '']
+        implicit_include_directories: bool
+        include_directories: T.List[IncludeDirs]
+        install: bool
+        install_dir: T.List[T.Union[str, Literal[False]]]
+        install_mode: FileMode
+        install_rpath: str
+        install_tag: T.List[str]
+        language_args: T.DefaultDict[str, T.List[str]]
```

Comment:
Why does it need to be a defaultdict?

Reply:
It doesn't technically have to be a defaultdict, but it would otherwise have to be something like:
```python
from mesonbuild.compilers import all_languages
v = {l: [] for l in all_languages}
```
Since we have several assumptions that directly indexing will not raise an exception: `BuildTarget.get_extra_args()`, and some of the code in the `extract_pic_pie` helper